core/src/main/java/com/scalar/db/storage/cosmos/SelectStatementHandler.java

```diff
@@ -89,22 +98,16 @@ private List<Record> executeRead(Get get, TableMetadata tableMetadata) throws Co
                 DSL.field("r.id").eq(cosmosOperation.getId()))
             .getSQL(ParamType.INLINED);
 
-    return getContainer(get).queryItems(query, new CosmosQueryRequestOptions(), Record.class)
-        .stream()
-        .collect(Collectors.toList());
+    return executeQuery(get, tableMetadata, query);
```

Comment:
I'm not sure but should we specify a partition key here?
```suggestion
    return executeQuery(get, tableMetadata, query, new CosmosQueryRequestOptions().setPartitionKey(cosmosOperation.getCosmosPartitionKey()));
```

Reply:
In the sdk [documentation](https://docs.microsoft.com/en-us/azure/cosmos-db/sql/sql-api-query-metrics#about-sql-query-execution) it says 

> When you issue a query to Azure Cosmos DB, the SDK performs these logical steps:
> ...
>     If the query includes a filter against the partition key, like SELECT * FROM c WHERE c.city = "Seattle", **it is routed to a single partition**. If the query does not have a the filter on partition key, then it is executed in all partitions, and results are merged client side.
> ...

So I assume that setting the partition key via the query options is redundant for this query as the where clause already filters on the partition key. This means the query will be routed to a single partition with or without the `CosmosQueryRequestOptions().setPartitionKey` anyway. 

Even though it does not constitute a benchmark, I executed from my laptop to the cosmos integration test machine the query with the added `CosmosQueryRequestOptions().setPartitionKey(cosmosOperation.getCosmosPartitionKey()` and did not notice a significant performance gain or loss compared to the original. A real benchmark would be necessary to see if it improves anything.

Reply:
I see. So we don't need to specify a partition key. Thank you for the investigation!